### PR TITLE
Enable `ember-suave`, start moving over to ES2015

### PR DESCRIPTION
**Pull request by duggiefresh**

### CODE REVIEW

**Comment by duggiefresh** [comment]
@bcardarella This PR uses the conventions as defined by `ember-suave`. Since it's pretty huge, let me know if you want me to break this into separate commits.

There is a spot that'll require some attention, https://github.com/dockyard/ember-validations/pull/300/files#diff-68a377996e559f04a75ca6ba1af084b1R167. That `var validatorName` isn't isolated only to the `for` loop, and bleeds into https://github.com/dockyard/ember-validations/pull/300/files#diff-68a377996e559f04a75ca6ba1af084b1R144.

The `for` loop should contain `let validatorName`, however, that means l think we'll have to somehow pass the name of the validator to the `pushValidator` constant. It's not a show stopper for now, but wanted to make a note of it.

File: addon/errors.js [hunk 1]
```diff
@@ -1,11 +1,14 @@
 import Ember from 'ember';
 
-var get = Ember.get;
-var set = Ember.set;
+const { A } = Ember;
 
-export default Ember.Object.extend({
-  unknownProperty: function(property) {
-    set(this, property, Ember.A());
+const EmberObject = Ember.Object;
```
Comment (bcardarella) [inline addon/errors.js]: Can't we just do:

```javascript
const {
  Object: EmberObject,
  get,
  set
} = Ember
```

**Comment by danmcclain** [comment]
We can't have get and set in destructure right now due to a bug in the version of esprima that jscs uses. Jscs can't parse the file if we do it right now

File: addon/errors.js (hunk 1 above)
Comment (rwjblue) [inline addon/errors.js]: The syntax is correct, but we cannot use destructuring with get and set at the moment (due to a bug in esprima used by JSCS).

**Comment by martndemus** [comment]
Was fixed in #423

**Pull request closed without merging**